src/main/java/solana/Solana.java

```diff
@@ -0,0 +1,382 @@
+                parsedInputAsArray = parsedInput.split(" \\(At: ", SPLIT_LIMIT);
+                newEvent = new Event(parsedInputAsArray[EVENT_INDEX],
+                        parsedInputAsArray[AT_INDEX].replace(")", ""));
+            }
+
+            tasks.add(newEvent);
+
+            if (isFromUser) {
+                printAddedPrompt(newEvent);
+                saveTasks();
+            }
+
+            if (isMarked) {
+                newEvent.markAsDone();
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Include the date or time using keyword \"/at\"!" + System.lineSeparator());
+        }
+    }
+
+    public static void deleteCommand(String parsedInput) {
+        try {
+            int taskIndex = Integer.parseInt(parsedInput);
+            printDeletedPrompt(taskIndex + CONVERT_TASK_TO_I);
+        } catch (NumberFormatException e) {
+            System.out.println("Input a task number!");
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            System.out.println("No such task number!");
+        }
+
+        saveTasks();
+        System.out.print(System.lineSeparator());
+    }
+
+    public static void parseInputFromFile(String[] taskAsArray) {
+        switch(taskAsArray[TASK_TYPE_INDEX]) {
+        case "[T][":
+            todoCommand(taskAsArray[DESCRIPTION_INDEX].substring(STARTING_DESCRIPTION_INDEX),
+                    false, false);
+            break;
+        case "[T][X]":
+            todoCommand(taskAsArray[DESCRIPTION_INDEX], false, true);
+            break;
+        case "[D][":
+            deadlineCommand(taskAsArray[DESCRIPTION_INDEX].substring(STARTING_DESCRIPTION_INDEX),
+                    false, false);
+            break;
+        case "[D][X]":
+            deadlineCommand(taskAsArray[DESCRIPTION_INDEX], false, true);
+            break;
+        case "[E][":
+            eventCommand(taskAsArray[DESCRIPTION_INDEX].substring(STARTING_DESCRIPTION_INDEX),
+                    false, false);
+            break;
+        case "[E][X]":
+            eventCommand(taskAsArray[DESCRIPTION_INDEX], false, true);
+            break;
+        default:
+            System.out.println("Unable to identify task type!");
+        }
+    }
+
+    public static void readFromFile(File savedTasks) throws FileNotFoundException {
+        Scanner in = new Scanner(savedTasks);
+        while (in.hasNext()) {
+            String taskAsString = in.nextLine().substring(TASK_INDEX);
+            String[] taskAsArray = taskAsString.split(" ", SPLIT_LIMIT);
+            parseInputFromFile(taskAsArray);
+        }
+    }
+
+    public static void loadTasks() {
+        File savedTasks = new File(FILE_PATH);
+        if (savedTasks.exists()) {
+            try {
+                readFromFile(savedTasks);
+            } catch (FileNotFoundException e) {
+                System.out.println("savedTasks.txt not found!");
+            }
+        }
```

Comment:
The structure of this code section has an almost arrowhead style. You might want to complement the `if` logic to separate it from the `try-catch` block. Doing this could also make the happy path more prominent.

Reply:
Good suggestion!